bitnami/redis/templates/master/statefulset.yaml

```
@@ -166,6 +166,22 @@ spec:
               containerPort: {{ .Values.master.containerPorts.redis }}
           {{- if not .Values.diagnosticMode.enabled }}
           {{- if .Values.master.livenessProbe.enabled }}
+          startupProbe:
+            initialDelaySeconds: {{ .Values.master.startupProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.master.startupProbe.periodSeconds }}
+            # One second longer than command timeout should prevent generation of zombie processes.
+            timeoutSeconds: {{ add1 .Values.master.startupProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.master.startupProbe.successThreshold }}
+            failureThreshold: {{ .Values.master.startupProbe.failureThreshold }}
```

The previous condition is wrong (line 168):
```diff
-{{- if .Values.master.livenessProbe.enabled }}
+{{- if .Values.master.startupProbe.enabled }}
```
Also.. use "omit":
```suggestion
          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.master.startupProbe "enabled") "context" $) | nindent 12 }}
```

---

What do you recommend me to do with this? 
```
            # One second longer than command timeout should prevent generation of zombie processes.
            timeoutSeconds: {{ add1 .Values.master.startupProbe.timeoutSeconds }}
```
Not sure what this workaround is intended for, but it would be lost if we use this method

---

Okay.. you can keep it that for the readiness/liveness probes, but I don't find it useful for the startup probe.
The startup probe shouldn't be using `ping_liveness_local.sh` nor `ping_readiness_local.sh` script, instead a simple "tcpSocket" probe should be enough. The goal of a startup probe is not checking the app's ready to accept connections or alive but just ensuring it's been started.